src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -154,17 +154,18 @@ export function CollapsibleNav({
     workspace: WorkspaceAttribute | null | undefined,
     categorizedLinks: Record<string, ChromeNavLink[]>
   ) {
-    // plugins are in this dictionary
-    const pluginsDictionary = categorizedLinks.opensearch;
-    if (!pluginsDictionary) return categorizedLinks;
-
     const features = workspace?.features ?? [];
-    const newPluginsDictionary = pluginsDictionary.filter((item) => features.indexOf(item.id) > -1);
-    if (newPluginsDictionary.length === 0) {
-      delete categorizedLinks.opensearch;
-    } else {
-      categorizedLinks.opensearch = newPluginsDictionary;
-    }
+
+    Object.keys(categorizedLinks).forEach((category) => {
+      const dictionary = categorizedLinks[category];
+      const newDictionary = dictionary.filter((item) => features.indexOf(item.id) > -1);
+      if (newDictionary.length === 0) {
+        delete categorizedLinks[category];
```

Comment:
Can we not mutate the original input?

Reply:
Yes, we can. I tried to modify it because I found that the original object is not used elsewhere. Do you recommend a deep copy way?

Reply:
I think you can just filter the `navLinks`, which is flattened link object list